lib/config.js

```diff
@@ -1653,17 +1673,25 @@ util.runStrictnessChecks = function (config) {
     _warnOrThrow("NODE_ENV value of '"+NODE_ENV+"' is ambiguous.");
   }
 
-  function _warnOrThrow (msg) {
-    var beStrict = process.env.NODE_CONFIG_STRICT_MODE;
-    var prefix = beStrict ? 'FATAL: ' : 'WARNING: ';
-    var seeURL = 'See https://github.com/lorenwest/node-config/wiki/Strict-Mode';
+  function _warnOrThrow (msg, neverThrow) {
+    var prefix = beStrict && !neverThrow ? 'FATAL: ' : 'WARNING: ';
 
     console.error(prefix+msg);
-    console.error(prefix+seeURL);
 
-    // Accept 1 and true as truthy values. When set via process.env, Node.js casts them to strings.
-    if (["true", "1"].indexOf(beStrict) >= 0) {
-      throw new Error(prefix+msg+' '+seeURL);
+    // Unless this is a warning that's never fatal, let the user know about strict mode.
+    if (!neverThrow) {
+      var modeInfo = beStrict ? 'Strict mode is on, aborting.' : 'Strict mode is off, continuing.';
+      var modeMessage =  modeInfo + ' See https://github.com/lorenwest/node-config/wiki/Strict-Mode';
+      console.error(prefix+modeMessage);
+      if (beStrict) {
+        throw new Error(prefix+msg+' '+modeMessage);
+      }
+    }
+
+    // If we're not in strict mode or this is a message that's safe to disable, let the user
+    // know that they can turn off warnings altogether
+    if (!beStrict || neverThrow) {
+      console.error(prefix+'To disable this warning, set SUPPRESS_NO_CONFIG_WARNING in the environment.');
```

Comment:
Okay, do you just want me to do this name change as part of this PR? I don't believe formal deprecation code with a deprecation period is required, since users who previously used the old env name will just see a notice telling them about the new env name, and the warning is only shown once on load so there's no risk of huge log pollution. But we'd still update history, wiki, etc. Does that make sense?